in_toto/gpg/functions.py

```diff
@@ -0,0 +1,168 @@
+"""
+<Module Name>
+  gpg/functions.py
+
+<Author>
+  Santiago Torres-Arias <santiago@nyu.edu>
+
+<Started>
+  Nov 15, 2017
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  publicly-usable functions for exporting public-keys, signing data and
+  verifying signatures.
+"""
+import subprocess
+import shlex
+
+import in_toto.log
+from in_toto.gpg.constants import (GPG_EXPORT_PUBKEY_COMMAND, GPG_SIGN_COMMAND,
+    SIGNATURE_HANDLERS)
+from in_toto.gpg.common import parse_signature_packet, parse_pubkey_packet
+
+import securesystemslib.formats
+
+
+def gpg_sign_object(content, keyid=None, homedir=None):
+  """
+  <Purpose>
+    Calls gpg command line utility to sign the passed content with the key
+    identified by the passed keyid from the gpg keyring at the passed homedir.
+
+  <Arguments>
+    content:
+            The content to be signed. (bytes)
+
+    keyid: (optional)
+            The keyid of the gpg signing keyid. If not passed the default
+            key in the keyring is used.
+            Note: On not fully supported gpg versions the keyid must be passed.
+
+    homedir: (optional)
+            Path to the gpg keyring. If not passed the default keyring is used.
+
+  <Exceptions>
+    None.
+
+  <Side Effects>
+    None.
+
+  <Returns>
+    The created signature in the format: gpg.formats.SIGNATURE_SCHEMA.
+  """
+
+  keyarg = ""
+  if keyid:
+    keyarg = "--default-key {}".format(keyid)
+
+  homearg = ""
+  if homedir:
+    homearg = "--homedir {}".format(homedir)
+
+  command = GPG_SIGN_COMMAND.format(keyarg=keyarg, homearg=homearg)
+  process = subprocess.Popen(shlex.split(command), stdout=subprocess.PIPE,
+      stdin=subprocess.PIPE, stderr=None)
+  signature_data, _ = process.communicate(content)
```

Comment:
We recently added a guideline for "junk" variables.
https://github.com/secure-systems-lab/code-style-guidelines#ignored-values